test/subscription_map.hpp

```diff
@@ -472,16 +480,24 @@ class multiple_subscription_map
 
         auto& subscription_set = h_iter->second.value;
 
-        auto new_pair = std::make_pair(key, std::forward<V>(value));
-        auto insert_result = subscription_set.insert(new_pair);
-        if (insert_result.second) {
+        // I have to do a find first, because insert_or_assign is available in C++17 only
+        // https://www.fluentcpp.com/2018/12/11/overview-of-std-map-insertion-emplacement-methods-in-cpp17/
+
+        auto existing_iter = subscription_set.find(key);
+        if(existing_iter == subscription_set.end()) {
+            subscription_set.insert(std::make_pair(std::forward<Key>(key), std::forward<V>(value)));
             ++this->map_size;
             this->increase_subscriptions(h);
         } else {
-            insert_result.first->second = new_pair.second;
+            existing_iter->second = std::forward<V>(value);
         }
 
-        return std::make_pair(h, insert_result.second);
+        return std::make_pair(h, existing_iter == subscription_set.end());
+    }
+
+    template<typename V>
+    std::pair<handle, bool> insert_or_update(handle h, Key const &key, V&& value) {
+        return insert_or_update(h, Key(key), std::forward<V>(value));
     }
```

Comment:
This version of the function isn't necessary.

Reply:
It is convenient, otherwise you would have to make a temporary when everytime you call insert_or_update with a const key.

Reply:
That doesn't make any sense.

What prevents std::forward<Key> from working normally?